src/FeedsMigrateImporterInterface.php

```diff
@@ -19,4 +19,142 @@ interface FeedsMigrateImporterInterface extends ConfigEntityInterface {
    */
   const SCHEDULE_CONTINUOUSLY = 0;
 
+  /**
+   * Indicates that existing items should be left alone.
+   */
+  const EXISTING_LEAVE = 'leave';
+
+  /**
+   * Indicates that existing items should be replaced.
+   */
+  const EXISTING_REPLACE = 'replace';
+
+  /**
+   * Indicates that existing items should be updated.
+   */
+  const EXISTING_UPDATE = 'update';
+
+  /**
+   * Get the import frequency.
+   *
+   * @return int
+   *   Interval in seconds.
+   */
+  public function getImportFrequency();
+
+  /**
+   * Set the import frequency.
+   *
+   * @param int $import_frequency
+   *   Interval in seconds.
+   */
+  public function setImportFrequency(int $import_frequency);
+
+  /**
+   * Get how existing items should be processed.
+   *
+   * @return string
+   *   One of the following:
+   *   - 'leave' => FeedsMigrateImporterInterface::EXISTING_LEAVE.
+   *   - 'replace' => FeedsMigrateImporterInterface::EXISTING_REPLACE.
+   *   - 'update' => FeedsMigrateImporterInterface::EXISTING_UPDATE.
+   */
+  public function getExisting();
+
+  /**
+   * Sets how existing items should be processed.
+   *
+   * @param string $existing
+   *   One of the following:
+   *   - 'leave' => FeedsMigrateImporterInterface::EXISTING_LEAVE.
+   *   - 'replace' => FeedsMigrateImporterInterface::EXISTING_REPLACE.
+   *   - 'update' => FeedsMigrateImporterInterface::EXISTING_UPDATE.
+   */
+  public function setExisting(string $existing);
+
+  /**
+   * Returns if orphaned items should be kept.
+   *
+   * @return bool
+   *   TRUE if orphans should be kept, FALSE if they can be deleted.
+   */
+  public function keepOrphans();
+
+  /**
+   * Sets if orphaned items should be kept.
+   *
+   * @param bool $keep_orphans
+   *   TRUE if orphans should be kept, FALSE if they can be deleted.
+   */
+  public function setKeepOrphans(bool $keep_orphans);
```

Comment:
Same as above, I refactored existing code and am not quite sure of the reasoning behind some of these functions. They are also not yet hooked up to anything.

Reply:
I assume it's about cleaning up previously imported items that are now no longer in the source.

In Feeds 7.x-2.x, in addition to deleting items, you can also choose for unpublishing them (in case of nodes) or blocking them (in case of users). In Feeds 8.x-3.x, besides deleting, you can choose to apply any non-configurable action plugin, since people using the D7 version reported that they wanted to have more options besides deleting/unpublishing.